spec/text-buffer-spec.js

```diff
@@ -2,13 +2,31 @@ const path = require('path')
 const TextBuffer = require('../src/text-buffer')
 
 describe('when a buffer is already open', () => {
+  const filePath = path.join(__dirname, 'fixtures', 'sample.js')
+  const buffer = new TextBuffer()
+
   it('replaces foo( with bar( using /\bfoo\\(\b/gim', () => {
-    const filePath = path.join(__dirname, 'fixtures', 'sample.js')
-    const buffer = new TextBuffer()
     buffer.setPath(filePath)
     buffer.setText('foo(x)')
     buffer.replace(/\bfoo\(\b/gim, 'bar(')
 
     expect(buffer.getText()).toBe('bar(x)')
   })
+
+  describe('should properly replace regex literals', () => {
+    it('replaces tstat_fvars()->curr_setpoint[HEAT_EN] = new_tptr->heat_limit; with tstat_set_curr_setpoint(HEAT_EN, new_tptr->heat_limit);', () => {
```

Comment:
It's not easy to understand what this test case is supposed to do. Can we simplify it?